### ISSUE TITLE
Fourier transform in the Style Extraction module

### ISSUE THREAD

**Issue by cocoz150** [ettof/Diveseg#3]
Dear author,thanks for your contribution！But I am confused that Which part of the code corresponds to the Fourier transform in the Style Extraction module?I have only seen some Convolution operations in the'class StyleExtractor',which is in the path 'Diveseg/mask2former/modeling/backbone/diveseg.py',Line 43-62：
class StyleExtractor(nn.Module):
    def __init__(self, em_dim=1024):
        super().__init__()
        self.conv1 = nn.Conv2d(3, em_dim, kernel_size=3, stride=2, padding=1)
        self.conv2 = nn.Conv2d(em_dim, em_dim, kernel_size=3, stride=1, padding=1)
        self.conv3 = nn.Conv2d(em_dim, em_dim, kernel_size=1, stride=1)
        self.relu = nn.ReLU(inplace=True)
        self.avgpool = nn.AdaptiveAvgPool2d(1)
    def forward(self, images_a):
        x = self.conv1(images_a)
        x = self.relu(x)
        x = self.conv2(x)
        x = self.relu(x)
        x = self.conv3(x)
        x = self.relu(x)
        x = self.avgpool(x)
        bs, dim, _, _ = x.shape
        x = x.view(bs, dim, -1).transpose(1, 2)
        return x

**Comment by ettof** [comment]
Thanks for your question! The Fourier transform in the Style Extraction module is not in the StyleExtractor class, but in lines 157-167 of ‘Diveseg/mask2former/data/dataset_mappers/coco_instance_new_baseline_dataset_mapper.py’

#get Fourier amplitude and phase
fre = np.fft.fft2(image, axes=(0, 1))
fre_a = np.abs(fre)
fre_p = np.angle(fre)

#set the amplitude a constant
constant = fre_p.mean()
fre_ = fre_a * np.e ** (1j * constant)
img_a = np.abs(np.fft.ifft2(fre_, axes=(0, 1)))
img_a = img_cut(img_a)
dataset_dict["image_a"] = torch.as_tensor(np.ascontiguousarray(img_a.transpose(2, 0, 1)))

**Comment by cocoz150** [comment]
thanks for your interruptions,it has been solved!
